src/main.tsx

```diff
@@ -1,29 +1,40 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
 import './index.css';
-import { CartItemsProvider } from './shared/context/CartItemsProvider.tsx';
-import { SelectedCartItemsProvider } from './shared/context/SelectedCartItemsProvider.tsx';
+import { CartItemsProvider } from './features/cart/context/CartItemsProvider.tsx';
+import { SelectedCartWithOrderProvider } from './shared/context/SelectedCartWithOrderProvider.tsx';
+import { CouponsProvider } from './features/coupon/context/CouponsProvider.tsx';
 import { RouterProvider } from 'react-router';
 import { router } from './app/routes/routes.tsx';
 
 async function enableMocking() {
-  const isLocalhost = location.hostname === 'localhost';
+  const shouldUseMSW = import.meta.env.VITE_USE_MSW === 'true';
+
+  if (!shouldUseMSW) {
+    return Promise.resolve();
+  }
 
   const { worker } = await import('./mocks/browser');
   return worker.start({
     serviceWorker: {
-      url: isLocalhost ? '/mockServiceWorker.js' : '/react-shopping-cart/mockServiceWorker.js',
+      url:
+        import.meta.env.NODE_ENV === 'production'
+          ? '/react-shopping-cart/mockServiceWorker.js'
+          : '/mockServiceWorker.js',
     },
     onUnhandledRequest: 'bypass',
   });
 }
+
 enableMocking().then(() => {
   ReactDOM.createRoot(document.getElementById('root')!).render(
     <React.StrictMode>
       <CartItemsProvider>
-        <SelectedCartItemsProvider>
-          <RouterProvider router={router} />
-        </SelectedCartItemsProvider>
+        <CouponsProvider>
+          <SelectedCartWithOrderProvider>
+            <RouterProvider router={router} />
+          </SelectedCartWithOrderProvider>
+        </CouponsProvider>
```

Comment:
[provider 구조 관련 코멘트 1번]

```suggestion
   <React.StrictMode>
      <CartItemsProvider>
        <CouponsProvider>
          <SelectedCartItemsProvider>
            <OrderProvider>
              <RouterProvider router={router} />
            </OrderProvider>
          </SelectedCartItemsProvider>
        </CouponsProvider>
      </CartItemsProvider>
    </React.StrictMode>
```

제가 상추의 말을 잘 이해하지 못한것일수도 있겠으나, 이렇게 나열하는것도 가능해보입니다.
`SelectedCartWithOrderProvider` 파일에서 2번 코멘트가 이어집니다...